internal/app/components/terms/termRecords.go

```diff
@@ -2,16 +2,25 @@ package terms
 
 import (
 	"github.com/gzcharleszhang/course-planner/internal/app/components/courses"
+	"github.com/rs/xid"
 )
 
+type TermRecordId string
+
 type TermRecord struct {
 	Term          Term
+	Id            TermRecordId
 	CourseRecords courses.CourseRecords
 }
 
+func newTermRecordId() TermRecordId {
+	return TermRecordId(xid.New().String())
+}
+
 func NewTermRecord(termName TermName, uwTermId int) *TermRecord {
 	return &TermRecord{
 		Term:          NewTerm(termName, uwTermId),
+		Id:						 newTermRecordId(),
```

Comment:
did you run `gofmt -s -d .` in the root directory?

Reply:
if you use GoLand as an IDE, you can also right-click a file, and then do Go Tools > gofmt

Reply:
or use `gofmt` on save 😏